lib/internal/Magento/Framework/DB/Ddl/Table.php

```diff
@@ -427,13 +428,23 @@ public function addForeignKey($fkName, $column, $refTable, $refColumn, $onDelete
             default:
                 $onDelete = self::ACTION_NO_ACTION;
         }
+        switch ($onUpdate) {
+            case self::ACTION_CASCADE:
+            case self::ACTION_RESTRICT:
+            case self::ACTION_SET_DEFAULT:
+            case self::ACTION_SET_NULL:
+                break;
+            default:
+                $onUpdate = self::ACTION_NO_ACTION;
+        }
 
         $this->_foreignKeys[$upperName] = [
             'FK_NAME' => $fkName,
             'COLUMN_NAME' => $column,
             'REF_TABLE_NAME' => $refTable,
             'REF_COLUMN_NAME' => $refColumn,
-            'ON_DELETE' => $onDelete
+            'ON_DELETE' => $onDelete,
+            'ON_UPDATE' => $onUpdate
```

Comment:
This was my point. Magento may handle its tables the way it wants, but it must provide a complete and consistent framework that allows developers to define and use their own data structures.

That's why I'm maintaining my point of view: it's a critical lack of the framework. In that case I would say: a bug.